chiselFrontend/src/main/scala/Chisel/When.scala

```diff
@@ -2,9 +2,12 @@
 
 package Chisel
 
+import scala.language.experimental.macros
+
 import internal._
 import internal.Builder.pushCommand
 import internal.firrtl._
+import internal.sourceinfo.{SourceInfo, WhenTransform}
 
 object when {  // scalastyle:ignore object.name
```

Comment:
WhenContext isn't actually sealed. We might not add an apply method, but that's not impossible.

In general, anything that doesn't return Unit gets a macro.